ls_colors.py

```diff
@@ -0,0 +1,168 @@
+from ranger.gui.colorscheme import ColorScheme
+import ranger.gui.color as style
+import ranger.gui.context
+import ranger.gui.widgets.browsercolumn
+from os import getenv
+
+ls_colors = getenv('LS_COLORS').split(':')
```

Comment:
when I unset LS_COLORS this throws an exception that's uncaught. (`getenv('LS_COLORS')` is None, so `getenv('LS_COLORS').split(':')` raises an AttributeError.)

I think you want to check if `getenv('LS_COLORS')` is `None` instead of `ls_colors`. Or catch the exception.

Reply:
Can you provide a default to `getenv`? Otherwise you can use `os.environ.get('LS_COLORS', default value)`. Imo that looks like the proper way to solve this problem.

Reply:
Had to look it up myself, `getenv('LS_COLORS', default='')` looks good. (NB: in two places in the file)

What I didn't realise earlier. It seems the uncaught exception makes ranger fall back to the default colors. Once I add the `default` value or exception handling in `class base(ColorScheme):` and when `LS_COLORS` is not set, then ranger becomes black&white.

I don't think this is what we want here (`ls --color` falls back to default colors when `LS_COLORS` is not set).

Reply:
So, if we want the default `LS_COLORS`, we should either let `ranger` automatically fall back to its default scheme, or we should manually read the default `dircolors` file or run the `dircolors` command without arguments to get the default colouring. Also, are we sure that is what `ranger` really does? (falling back to default `LS_COLORS`) This could simply be a matter of catching the exception.

I've had trouble however finding the system wide config file that `dircolors` automatically reads. The man page (`man dir_colors`) points to either `~/.dircolors` (user-defined) or `/etc/DIR_COLORS` (system-wide). I, however, do not have the system-wide one and yet `dircolors` is still able to read the colours from somewhere. 

Have you had any luck finding whence it gets the colours?

Reply:
Not sure what's the best to do. I guess given the name 'ls_colors' it should be as close as reasonable to "what ls does".

ranger gets its default colors from https://github.com/ranger/ranger/blob/master/ranger/colorschemes/default.py which happens to be the same as (my?) ls defaults when LS_COLORS is unset.

I also don't see where my `dircolors` gets its defaults from.

Given we don't seem to see where `dircolors` gets its config from (and one would need to check user file, system file, fallback) I think calling `dircolors` if `LS_COLORS` is unset is a good handling.

Reply:
I added a commit to handle the case where `LS_COLORS` isn't set.  
Scenario cases: 
1) `LS_COLORS` is defined and parsed
2)  `LS_COLORS` isn't defined, `dircolors` is called and we use its output to define the `LS_COLORS` for the colorscheme
3) `LS_COLORS` isn't defined and the `dircolors` command can't be found. We use a black & white colorscheme. 

the 3) is probably not what the user want, but it might be a good idea to let them know something is wrong if neither a `LS_COLORS` env variable or a `dircolors` command can be found while using the `ls_colors.py` scheme.

Reply:
Ranger probably "falls back", actually doesn't change to ls_colors.py, because that throws an error.
The proper behavior would be to do what ls does. The `default.py` colorscheme doesn't completely match what `ls --color` does without `LS_COLORS` set.

I meant to use whatever actually is the default value for `LS_COLORS` as the default but calling dircolors every time is probably fine too.

Do you really need to duplicate the try/except block for the getenv?

Reply:
@toonn 

> Do you really need to duplicate the try/except block for the getenv?

Can you elaborate? I don't see duplication here. Or should this thread be resolved?

Reply:
This refers to an old version. the old file I still have from September has two calls `getenv('LS_COLORS')`. The current version in this PR does not have that duplication.